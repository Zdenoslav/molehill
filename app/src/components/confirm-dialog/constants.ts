export enum ConfirmDialogType {
  WARNING = 'WARNING',
  INFO = 'INFO',
}
