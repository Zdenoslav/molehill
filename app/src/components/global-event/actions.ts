import {createActions} from 'app/redux/create-actions';

const domain = 'GLOBAL_EVENT';

export const {
  action: updateError,
} = createActions(
  `${domain}/ERROR`,
  (message: string) => ({message}),
);

export const {
  action: dismissEvent,
} = createActions(
  `${domain}/DISMISS`,
  () => null,
);
