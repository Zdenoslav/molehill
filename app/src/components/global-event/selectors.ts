import { IRootState } from 'app/redux/root-reducer';

export const eventMessage = (state: IRootState) => state.globalEvent.message;
export const eventType = (state: IRootState) => state.globalEvent.type;
