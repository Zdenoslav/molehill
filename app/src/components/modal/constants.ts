export enum ModalIds {
  addNewStatus = 'addNewStatus',
}
