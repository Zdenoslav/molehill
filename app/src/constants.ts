// temp constants for development

export const RADIUS = 20; // km
