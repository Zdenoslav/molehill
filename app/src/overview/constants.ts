export const RADIUS_MAX = 30;
export const RADIUS_MIN = 1;
export const RADIUS_DEFAULT = 30;
