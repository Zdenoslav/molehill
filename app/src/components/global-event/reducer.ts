import {Record} from 'immutable';

import { IReduxAction } from 'app/redux/create-actions';

import * as Actions from './actions';

export enum GlobaEventType {
  ERROR = 'ERROR',
  SUCCESS = 'SUCCESS',
}

export interface IGlobalEvent {
  message: string | null;
  type: GlobaEventType | null;
}

export class GlobalEventState extends Record<IGlobalEvent>({
  message: null,
  type: null,
}) {}

const loginReducer = (state = new GlobalEventState(), action: IReduxAction) => {
  switch (action.type) {
    case Actions.updateError.type:
      return state.merge({
        message: action.payload.message,
        type: GlobaEventType.ERROR,
      });
    case Actions.dismissEvent.type:
      return new GlobalEventState();
    default:
      return state;
  }
};

export default loginReducer;
