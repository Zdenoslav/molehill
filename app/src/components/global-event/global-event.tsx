import React from 'react';
import { compose } from 'redux';
import { connect } from 'react-redux';

import { IRootState } from 'app/redux/root-reducer';
import styled from 'app/components/styleguide';
import { s2 } from '../styleguide/spacing';

import { Body, Title } from '../styleguide/text';
import {Flex} from '../styleguide/layout';

import {IGlobalEvent} from './reducer';
import {getGlobalEvent} from './selectors';
import {dismissEvent} from 'app/redux/internals';
import { GlobaEventType } from './constants';

export const DismissButton = styled(Title)`
  transform: rotate(45deg);
`;

interface IStateProps {
  globalEvent: IGlobalEvent;
}

interface IEventContainerProps {
  eventType: GlobaEventType;
}

export const GlobalEventContainer = styled.div<IEventContainerProps>`
  position: fixed;
  top: 40px;
  right: 40px;
  border-radius: 7px;
  padding: 12px 16px;
  z-index: 9998;
  background: ${props => {
    if (props.eventType === GlobaEventType.SUCCESS) {
      return `linear-gradient(to bottom right, ${props.theme.submitLight}, ${props.theme.submitDark})`;
    }
    else if (props.eventType === GlobaEventType.ERROR) {
      return `linear-gradient(to bottom right, ${props.theme.errorLight}, ${props.theme.errorDark})`;
    }
    return;
  }};
`;

export const GlobalEvent: React.SFC<IStateProps> = (props) => {
  const {
    globalEvent,
  } = props;

  const {
    message,
    type,
    sticky,
  } = globalEvent;

  if (!message || !type) {
    return null;
  }

  return (
    <GlobalEventContainer eventType={type}>
      <Flex align="center">
        <Body inverted>
          {message}
        </Body>
        {sticky && (
          <DismissButton onClick={dismissEvent.dispatch} clickable inverted marginLeft={s2}>+</DismissButton>
        )}
      </Flex>
    </GlobalEventContainer>
  );
};

export default compose<React.ComponentType>(
  connect<IStateProps, {}, {}, IRootState>((state) => ({
    globalEvent: getGlobalEvent(state),
  })),
)(GlobalEvent);
