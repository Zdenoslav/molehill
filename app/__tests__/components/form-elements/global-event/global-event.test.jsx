import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer'
import 'jest-styled-components';

import {GlobalEvent, GlobalEventContainer, DismissButton} from 'app/components/global-event/global-event';
import { GlobaEventType } from 'app/components/global-event/constants';
import Theme from 'app/components/styleguide/theme';

describe('GlobalEvent', () => {

  it('should render GlobalEvent', () => {
    const mockGlobalEvent = {
      message: 'Mock Global Event',
      type: GlobaEventType.SUCCESS,
      sticky: false,
    };
    const wrapper = shallow(<GlobalEvent globalEvent={mockGlobalEvent} />);
    expect(wrapper).toMatchSnapshot();
  });

  it(`should not render GlobalEvent if event doesn't contain type`, () => {
    const mockGlobalEvent = {
      message: 'Mock Global Event',
      sticky: false,
    };
    const wrapper = shallow(<GlobalEvent globalEvent={mockGlobalEvent} />);
    expect(wrapper).toMatchSnapshot();
  });

  it(`should not render GlobalEvent if event doesn't contain message`, () => {
    const mockGlobalEvent = {
      sticky: false,
      type: GlobaEventType.SUCCESS,
    };
    const wrapper = shallow(<GlobalEvent globalEvent={mockGlobalEvent} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Success GlobalEventContainer', () => {
    const wrapper = renderer.create(<GlobalEventContainer theme={Theme} eventType={GlobaEventType.SUCCESS} />).toJSON();
    expect(wrapper).toHaveStyleRule('background', `linear-gradient(to bottom right,${Theme.submitLight},${Theme.submitDark})`);
  });

  it('should render Error GlobalEventContainer', () => {
    const wrapper = renderer.create(<GlobalEventContainer theme={Theme} eventType={GlobaEventType.ERROR} />).toJSON();
    expect(wrapper).toHaveStyleRule('background', `linear-gradient(to bottom right,${Theme.errorLight},${Theme.errorDark})`);
  });

  it('should render default GlobalEventContainer', () => {
    const wrapper = renderer.create(<GlobalEventContainer theme={Theme} />).toJSON();
    expect(wrapper).toHaveStyleRule('background', undefined);
  });
  
  it('should render DismissButton', () => {
    const wrapper = shallow(<DismissButton />);
    expect(wrapper).toMatchSnapshot();
  });

});
