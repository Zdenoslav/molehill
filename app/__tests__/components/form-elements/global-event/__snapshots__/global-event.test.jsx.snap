// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalEvent should not render GlobalEvent if event doesn't contain message 1`] = `""`;

exports[`GlobalEvent should not render GlobalEvent if event doesn't contain type 1`] = `""`;

exports[`GlobalEvent should render DismissButton 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Object {
        "as": "p",
      },
      "componentStyle": ComponentStyle {
        "componentId": "sc-ckVGcZ",
        "isStatic": false,
        "rules": Array [
          "
      margin-top: ",
          [Function],
          ";
      margin-bottom: ",
          [Function],
          ";
      margin-left: ",
          [Function],
          ";
      margin-right: ",
          [Function],
          ";
    
      padding-top: ",
          [Function],
          ";
      padding-bottom: ",
          [Function],
          ";
      padding-left: ",
          [Function],
          ";
      padding-right: ",
          [Function],
          ";
    
      flex-grow: ",
          [Function],
          ";
      width: ",
          [Function],
          ";
    
      cursor: ",
          [Function],
          ";
    ",
          "
      text-align: ",
          [Function],
          ";
      color: ",
          [Function],
          ";
      font-weight: ",
          [Function],
          ";
      opacity: ",
          [Function],
          ";
      word-break: ",
          [Function],
          ";
    
      ",
          [Function],
          "
    ",
          "
      font-size: 20px;
    ",
          "
      transform: rotate(45deg);
    ",
        ],
      },
      "displayName": "Styled(Styled(Styled(styled.div)))",
      "render": [Function],
      "styledComponentId": "sc-ckVGcZ",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`GlobalEvent should render GlobalEvent 1`] = `
<styled.div
  eventType="SUCCESS"
>
  <Styled(styled.div)
    align="center"
  >
    <Styled(Styled(styled.div))
      inverted={true}
    >
      Mock Global Event
    </Styled(Styled(styled.div))>
  </Styled(styled.div)>
</styled.div>
`;
